pyboy/gymboy.py

```diff
@@ -0,0 +1,165 @@
+                - 'press' the agent will only press inputs for 1 frame an then release it.
+                - 'toggle' the agent will toggle inputs, first time it press and second time it release.
+                - 'all' the agent have acces to all inputs, press and release are separated.
+        
+        game_wrapper: `pyboy.plugins.base_plugin.PyBoyGameWrapper`
+            The game_wrapper of the PyBoy game instance over which the environment is built.
+
+        action_space: Gym space
+            The action space of the environment.
+
+        observation_space: Gym space
+            The observation space of the environment (depends of observation_type).
+        
+        last_fitness: float or int
+            The last observed fitness, used to compute the reward at each step
+
+        actions: list
+            The list of input IDs of allowed input for the agent (depends of buttons_press_mode).
+    
+    """
+
+    def __init__(self, game, observation_type='tiles', buttons_press_mode='toggle', simultaneous_actions=False):
+        # Build game
+        self.game = game
+        if str(type(game)) != "<class 'pyboy.pyboy.PyBoy'>":
+            raise TypeError("game must be a Pyboy object")
+        
+        # Build game_wrapper
+        self.game_wrapper = game.game_wrapper()
+        if self.game_wrapper is None:
+            raise ValueError("You need to build a game_wrapper to use this function. Otherwise there is no way to build a reward function automaticaly (for now).")
+        self.last_fitness = self.game_wrapper.fitness
+
+        # Building the action_space
+        self._DO_NOTHING = -1
+        self._buttons = list(range(1, 9)) # UP DOWN RIGHT LEFT A B SELECT START
+        self._button_is_pressed = {button:False for button in self._buttons}
+        
+        self._buttons_release = list(range(9, 17)) # UP DOWN RIGHT LEFT A B SELECT START
+        self._release_button = {button:r_button for button, r_button in zip(self._buttons, self._buttons_release)}
+        
+        self.actions = [self._DO_NOTHING] + self._buttons
+        if buttons_press_mode == 'all':
+            self.actions += self._buttons_release
+        elif not (buttons_press_mode == 'press' or buttons_press_mode == 'toggle'):
+            raise ValueError(f'buttons_press_mode {buttons_press_mode} is unknowed')
+        self.buttons_press_mode = buttons_press_mode
+
+        if simultaneous_actions:
+            raise NotImplementedError("Not implemented yet, raise an issue if needed")
+        else:
+            self.action_space = Discrete(len(self.actions))
+
+        # Building the observation_space
+        if observation_type == 'raw':
+            screen = np.asarray(self.game.botsupport_manager().screen().screen_ndarray())
+            self.observation_space = Box(low=0, high=255, shape=screen.shape, dtype=np.uint32)
+        elif observation_type == 'tiles':
+            nvec = 384 * np.ones(self.game_wrapper.shape)
```

Comment:
> 384

Could be `pyboy.botsupport.constants.TILES` instead of a "magic number"

Reply:
Sure ! Did not know that this constant was there !